apps/prairielearn/src/sync/sharing.ts

```diff
@@ -186,3 +186,39 @@ export async function checkInvalidSharingSetRemovals(
 
   return existInvalidSharingSetRemovals;
 }
+
+export function checkInvalidSharedAssessments(
+  courseData: CourseData,
+  logger: ServerJobLogger,
+): boolean {
+  const invalidSharedAssessments = new Set<string>();
+  for (const courseInstanceKey in courseData.courseInstances) {
+    const courseInstance = courseData.courseInstances[courseInstanceKey];
+    for (const tid in courseInstance.assessments) {
+      const assessment = courseInstance.assessments[tid];
+      if (!assessment?.data?.shareSourcePublicly) {
+        continue;
+      }
+      for (const zone of assessment?.data?.zones ?? []) {
+        for (const question of zone.questions ?? []) {
+          if (!question.id) {
+            continue;
+          }
+          const infoJson = courseData.questions[question.id];
+          if (!infoJson?.data?.sharePublicly) {
```

Comment:
This could benefit from a comment explaining why `shareSourcePublicly` isn't considered.

Reply:
I actually do have a question on this. Should we have both because `sharePublicly` means the question can be imported, and `shareSourcePublicly` means the source code can be copied. Doesn't this mean that `shareSourcePublicly` is like a stronger version of `sharePublicly` and we should allow the question to be imported even if only `shareSourcePublicly` is true?

Reply:
`sharePublicly` (allow import) and `shareSourcePublicly` (allow copy) are mutually exclusive. If only `shareSourcePublicly` is true, the question isn't importable via `@sharing-name/qid`. Using such a question in a shared assessment would mean we'd also have to copy the question into the destination course, which I think we wanted to avoid for now.

@SethPoulsen does this match what we discussed previously?

Reply:
Yes, that's correct.

Reply:
comment has been added on line 211